test/unit/org/apache/cassandra/repair/ValidationTaskTest.java

```diff
@@ -52,10 +52,12 @@ public void shouldIgnoreTreesWhenDeactivated() throws Exception
     {
         ValidationTask task = createTask();
         assertTrue(task.isActive());
-        task.abort();
+        task.abort(new RuntimeException());
         assertFalse(task.isActive());
         task.treesReceived(new MerkleTrees(null));
-        assertNull(task.get());
+        // REVIEW: setting null would cause NPEs in sync task, so it was never correct to set null
+        assertTrue(task.isDone());
+        assertFalse(task.isSuccess());
```

Comment:
So a `null` `TreeResponse` could arrive at, for instance, `createStandardSyncTasks()` and be dereferenced?

Reply:
before, when we abort we didn't resolve the future and had a boolean to track that, so if we abort then we would eventually resolve as `null`, which doesn't actually stop the repair future chain, so `org.apache.cassandra.repair.asymmetric.DifferenceHolder#DifferenceHolder(java.util.List<org.apache.cassandra.repair.TreeResponse>)` will get called and NPE at 

```
hd.add(r2.endpoint, MerkleTrees.difference(r1.trees, r2.trees));
```

Since the job already aborted this will report correctly to ops, but "could" get a NPE in the logs